packages/ui/src/customRenderers/fileUpload/fileUpload.tsx

```diff
@@ -40,27 +51,40 @@ const FileUpload = forwardRef(function FileUpload(props: FileUploadProps, ref) {
 					fileUploadUrl: file.response?.uploadURL,
 					filePreview: file.preview,
 				};
-			});
+			}) as FormattedFile[];
 			props.onUpdateFiles(formattedFiles);
 		});
 	}, [props.onUpdateFiles]);
 
-	uppy.getPlugin<AwsS3Multipart<Meta, AwsBody>>(pluginName)!.setOptions({
-		// TODO: maybe use more specific types for Meta and Body
-		getUploadParameters: async (file) => {
-			if (!file || !file.type) {
-				throw new Error("Could not read file.");
-			}
-			const signedUrl = await props.upload(file.name);
-			return {
-				method: "PUT",
-				url: signedUrl,
-				headers: {
-					"content-type": file.type,
-				},
-			};
-		},
-	});
+	useEffect(() => {
+		uppy.getPlugin<AwsS3Multipart<Meta, AwsBody>>(pluginName)!.setOptions({
+			// TODO: maybe use more specific types for Meta and Body
+			getUploadParameters: async (file) => {
+				if (!file || !file.type) {
+					throw new Error("Could not read file.");
+				}
+
+				if (!file.name) {
+					throw new Error("File name is required");
+				}
+
+				const signedUrl = await props.upload(file.name);
+
+				if (typeof signedUrl === "object" && "error" in signedUrl) {
+					throw new Error(signedUrl.error);
+				}
+
+				return {
+					method: "PUT",
+					url: signedUrl,
+					headers: {
+						"content-type": file.type,
+					},
+				};
+			},
+		});
+	}, [props.upload]);
+
```

Comment:
this does not really change the logic (adds two extra checks but that's it)

this should be wrapped in an `effect` though